File: 2-й курс/Нечеткие модели/Lab. 1/with.m
clear all; close all; clc;

X = 60:1:220;

hold on

plot(X, zmf(X,[80,100]))
plot(X, pimf(X,[100,115,125,140]))
plot(X, smf(X,[140,160]))
plot(X, 1 - zmf(X,[80,100]).^2)
plot(X, pimf(X,[100,115,125,140]).^(1/2))
plot(X, smf(X,[140,160]).^2)

legend("Низький","Помірний","Високий","Не дуже низький", "Більш-менш помірний","Дуже високий")

grid on
xlim([min(X) max(X)])
xlabel('Тиск')
ylabel('Належність')

title('Використовуючи внутрішні функції')

File: 2-й курс/Нечеткие модели/Lab. 1/with.py
import numpy as np
import matplotlib.pyplot as plt
import skfuzzy as fuzz

X = np.arange(60, 220)

fig, ax = plt.subplots()

ax.set_xlabel("Тиск")
ax.set_ylabel("Належність")
ax.set_title("З використанням бібліотеки skfuzzy")

ax.plot(X, fuzz.zmf(X, 80, 100), label="Низький")
ax.plot(X, fuzz.pimf(X, 100, 115, 125, 140), label="Помірний")
ax.plot(X, fuzz.smf(X, 140, 160), label="Високий")

ax.plot(X, 1 - fuzz.zmf(X, 80, 100)**2, label="Не дуже низький")
ax.plot(X, fuzz.pimf(X, 100, 115, 125, 140)
        ** (1/2), label="Більш-менш помірний")
ax.plot(X, fuzz.smf(X, 140, 160)**2, label="Дуже високий")


ax.legend()

plt.grid()
plt.show()
